partiql-plan/src/main/java/org/partiql/plan/rex/RexBag.java

```diff
@@ -0,0 +1,62 @@
+package org.partiql.plan.rex;
+
+import org.jetbrains.annotations.NotNull;
+import org.partiql.plan.Operator;
+import org.partiql.plan.OperatorVisitor;
+import org.partiql.types.PType;
+
+import java.util.Collection;
+import java.util.List;
+
+/**
+ * Logical bag expression abstract base class.
+ */
+public abstract class RexBag extends RexBase {
+
+    /**
+     * @return new RexBag instance
+     */
+    @NotNull
+    public static RexBag create(@NotNull Collection<Rex> values) {
+        return new Impl(values);
+    }
+
+    /**
+     * @return the values of the bag, also the operands (unordered).
+     */
+    @NotNull
+    public abstract Collection<Rex> getValues();
+
+    @NotNull
+    @Override
+    protected final RexType type() {
+        return RexType.of(PType.bag());
+    }
+
+    @NotNull
+    @Override
+    protected final List<Operator> operands() {
+        Collection<? extends Operator> varargs = getValues();
+        return List.copyOf(varargs);
```

Comment:
While it probably doesn't pertain to this specific class (since I doubt additional fields would be added), I'm curious how we could mitigate breaking changes if we were to add another field to a class which has variable arguments. One way would be to create an OperatorList which extends Operator. That way, for the `operands()` method, you can still add elements (potentially variable as well) to the end of the list without ambiguity.

Reply:
Yes I have a note in #1664 about handling varargs and just using a holder works, but would require a larger refactor for what an "operand" is. I think having operands mirror patterns would be easiest here, which would mean having a special vararg "operand".